geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/misc/WANSSLDUnitTest.java

```diff
@@ -14,107 +14,624 @@
+  public int dispatcherThreadCount;
+
+  /** This VM's connection to the distributed system */
+  private static InternalDistributedSystem system;
+  private String testMethodName;
+  private String replicatedRegionName;
+
+  @Rule
+  public SerializableTestName testMethodNameWatcher = new SerializableTestName();
+
+  @Rule
+  public ClusterStartupRule csRule = new ClusterStartupRule();
+
+  @Before
+  public void before() {
+    sendingLocator = csRule.getVM(0);
+    receivingLocator = csRule.getVM(1);
+    receivingServer = csRule.getVM(2);
+    sendingServer = csRule.getVM(3);
+
+    sendingLocatorPort = sendingLocator.invoke(() -> createFirstLocatorWithDSIdOne());
+    receivingLocatorPort = receivingLocator.invoke(() -> createFirstRemoteLocatorWithDSIdTwo(sendingLocatorPort));
+
+    // setup distributed test case
+    // Make sure all VMs agree on nomenclature
+    final String className = getClass().getCanonicalName();
+    testMethodName = getSanitizedCurrentTestName();
+    replicatedRegionName = testMethodName + "_RR";
+    String methodName = testMethodName;
+    setUpVM(getDefaultDiskStoreName(0, -1, className, testMethodName));
+
+    for (int hostIndex = 0; hostIndex < Host.getHostCount(); hostIndex++) {
+      Host host = Host.getHost(hostIndex);
+      for (int vmIndex = 0; vmIndex < host.getVMCount(); vmIndex++) {
+        final String vmDefaultDiskStoreName =
+            getDefaultDiskStoreName(hostIndex, vmIndex, className, testMethodName);
+        host.getVM(vmIndex).invoke("setupVM", () -> {
+          testMethodName = methodName;
+          replicatedRegionName = methodName + "_RR";
+          setUpVM(vmDefaultDiskStoreName);
+        });
+      }
+    }
 
-  public WANSSLDUnitTest() {
-    super();
+    // setup WAN test case
+    int nThreads = dispatcherThreadCount;
+    Invoke.invokeInEveryVM(() -> dispatcherThreadCount = nThreads);
+
+    IgnoredException.addIgnoredException("Connection refused");
+    IgnoredException.addIgnoredException("Software caused connection abort");
+    IgnoredException.addIgnoredException("Connection reset");
   }
 
-  @Test
-  public void testSenderSSLReceiverSSL() {
-    Integer lnPort = (Integer) vm0.invoke(() -> WANTestBase.createFirstLocatorWithDSId(1));
-    Integer nyPort = (Integer) vm1.invoke(() -> WANTestBase.createFirstRemoteLocator(2, lnPort));
+  private Integer createFirstLocatorWithDSIdOne() {
+    int port = AvailablePortHelper.getRandomAvailablePortForDUnitSite();
+    startLocator(1, port, port, -1);
+    return port;
+  }
 
-    vm2.invoke(() -> WANTestBase.createReceiverWithSSL(nyPort));
+  private Integer createFirstRemoteLocatorWithDSIdTwo(int remoteLocPort) {
+    int port = AvailablePortHelper.getRandomAvailablePortForDUnitSite();
+    startLocator(2, port, port, remoteLocPort);
+    return port;
+  }
 
-    vm4.invoke(() -> WANTestBase.createCacheWithSSL(lnPort));
+  private void createReplicatedRegion(String senderIds) {
+    IgnoredException exp =
```

Comment:
You could change this to use try-resource instead of try-finally and also no need to specify class.getName():
```
import static org.apache.geode.test.dunit.IgnoredException.addIgnoredException;

try (IgnoredException ie1 = addIgnoredException(ForceReattemptException.class);
    IgnoredException ie2 = addIgnoredException(InterruptedException.class);
    IgnoredException ie3 = addIgnoredException(GatewaySenderException.class)) {
...
}
```